webapp/src/test/java/com/box/l10n/mojito/service/screenshot/ScreenshotServiceTest.java

```diff
@@ -150,7 +150,7 @@ public void testDeleteScreenshotNoThirdPartyConfig() throws RepositoryNameAlread
     tmTextUnitRepository.save(tmTextUnit);
 
     ScreenshotTextUnit screenshotTextUnit = new ScreenshotTextUnit();
-    screenshotTextUnit.setId(1L);
+    screenshotTextUnit.setId(2L);
```

Comment:
Why is this change needed?

Reply:
This test was failing with that id

Reply:
Before the changes, the screenshot variable had an ID of 29. However, after the changes, its ID remains 1 (the record with that id was removed here and it's causing the issue).
<img width="1155" alt="Screenshot 2024-07-01 at 9 08 49 PM" src="https://github.com/pinterest/mojito/assets/40587669/59a39feb-b251-43e8-8577-dc2152c6e777">

Reply:
Not sure I follow? The test deletes a screenshot and verifies it has been deleted, what error do you see when it's left as is?

Reply:
For the current changes, the code below prints "1 - 29" for test **testDeleteScreenshotNoThirdPartyConfig**
<img width="1259" alt="Screenshot 2024-07-02 at 11 44 50 AM" src="https://github.com/pinterest/mojito/assets/40587669/3acda852-63b6-421f-8c6b-c5f06462a4fd">
Before the changes, it printed "29 - 29". That record (with ID = 1) was removed in test testDeleteScreenshot, and that is why the assert `Assert.assertNotNull(screenshotRepository.findById(screenshot.getId()).orElse(null));` threw an exception

Reply:
This was changed to the following:
<img width="948" alt="Screenshot 2024-07-03 at 7 28 15 PM" src="https://github.com/pinterest/mojito/assets/40587669/e8e8d82e-e428-4905-b6bb-b91cc97923d8">

Reply:
Is your database being cleaned before the test execution? It sounds like the id mismatch is due to you already having screenshots existing in the test database before the test is run. Below  in your application.properties file will clean the test database prior to test execution.

Ensure to comment it out when running some manual testing locally as it will clean your dev db in that scenario


`l10n.flyway.clean=true
spring.flyway.cleanDisabled=false
spring.flyway.clean-disabled=false`

Reply:
My issue here is finding out if this is caused by the Spring upgrade or is it purely environmental, if it's the Spring upgrade what's changed to cause this behaviour and will it affect us elsewhere?

Reply:
@DarKhaos missed this was outdated, new code looks fine (running the tests against a fresh db locally is important though)